simulator-core/src/test/java/com/evbox/everon/ocpp/simulator/station/handlers/ocpp/CancelReservationRequestHandlerTest.java

```diff
@@ -0,0 +1,88 @@
+package com.evbox.everon.ocpp.simulator.station.handlers.ocpp;
+
+import com.evbox.everon.ocpp.simulator.station.StationMessageSender;
+import com.evbox.everon.ocpp.simulator.station.StationStore;
+import com.evbox.everon.ocpp.simulator.station.evse.CableStatus;
+import com.evbox.everon.ocpp.simulator.station.evse.Connector;
+import com.evbox.everon.ocpp.v20.message.common.Evse;
+import com.evbox.everon.ocpp.v20.message.station.CancelReservationRequest;
+import com.evbox.everon.ocpp.v20.message.station.CancelReservationResponse;
+import com.evbox.everon.ocpp.v20.message.station.Reservation;
+import com.evbox.everon.ocpp.v20.message.station.StatusNotificationRequest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class CancelReservationRequestHandlerTest {
+
+    private static final String CALL_ID = "123";
+    private static final int RESERVATION_ID = 123;
+    private static final int CONNECTOR_ID = 1;
+    private static final int EVSE_ID = 2;
+
+    @Mock
+    private StationMessageSender stationMessageSender;
+
+    @Mock
+    private StationStore stationStore;
+
+    @InjectMocks
+    private CancelReservationRequestHandler handler;
+
+    @Captor
+    private ArgumentCaptor<CancelReservationResponse> cancelReservationResponseArgumentCaptor = ArgumentCaptor.forClass(CancelReservationResponse.class);
+
+    @Test
+    public void testCancelReservationRejected() {
+        assertCancelReservationStatus(CancelReservationResponse.Status.REJECTED);
+    }
+
+    @Test
+    public void testCancelReservationAcceptedWithNoConnectorId() {
+        Reservation reservation = buildReservation(EVSE_ID, RESERVATION_ID, null);
+
+        when(stationStore.tryFindReservationById(anyInt())).thenReturn(Optional.of(reservation));
+
+        assertCancelReservationStatus(CancelReservationResponse.Status.ACCEPTED);
+    }
+
+    @Test
+    public void testCancelReservationAcceptedWithConnectorId() {
+        Reservation reservation = buildReservation(EVSE_ID, RESERVATION_ID, CONNECTOR_ID);
+        Connector connector = new Connector(CONNECTOR_ID, CableStatus.PLUGGED, StatusNotificationRequest.ConnectorStatus.RESERVED);
+
+        when(stationStore.tryFindReservationById(anyInt())).thenReturn(Optional.of(reservation));
+        when(stationStore.tryFindConnector(anyInt(), anyInt())).thenReturn(Optional.of(connector));
+
+        assertCancelReservationStatus(CancelReservationResponse.Status.ACCEPTED);
```

Comment:
Are we asserting too much in one test? Is there a case for splitting this into two tests? I am not mandating that you must, just give it some thought

Reply:
Well, this is basically one flow when reservation also has a connector reserved. So, I don't think it should be splitted.